corehq/apps/api/resources/v0_5.py

```diff
@@ -497,6 +504,97 @@ def dehydrate(self, bundle):
         return bundle
 
 
+ConfigurableReportData = namedtuple("ConfigurableReportData", [
+    "data", "id", "start", "limit", "domain", "total_records", "get_params"
+])
+
+
+class ConfigurableReportDataResource(HqBaseResource, DomainSpecificResourceMixin):
+    """
+    A resource that replicates the behavior of the ajax part of the
+    ConfigurableReport view.
+    """
+    data = fields.ListField(attribute="data", readonly=True)
+    total_records = fields.IntegerField(attribute="total_records", readonly=True)
+
+
+    def obj_get(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        pk = kwargs['pk']
+        try:
+            start = int(kwargs['start'])
+            limit = int(kwargs['limit'])
+        except (ValueError, TypeError) as e:
+            raise BadRequest("start and limit values must be integers.")
+
+        if limit > 50:
+            raise BadRequest("Limit may not exceed 50.")
+
+        report_config = self._get_report_configuration(pk, domain)
+        report = ReportFactory.from_spec(report_config)
+
+        filter_values = get_filter_values(
+            report_config.ui_filters,
+            query_dict_to_dict(bundle.request.GET, domain)
+        )
+        report.set_filter_values(filter_values)
+
+        page = list(report.get_data(start=start, limit=limit))
+        total_records = report.get_total_records()
+
+        return ConfigurableReportData(
+            data=page,
+            total_records=total_records,
+            id=report_config._id,
+            domain=domain,
+            start=start,
+            limit=limit,
+            get_params=bundle.request.GET,
+        )
+
+    def _get_report_configuration(self, id_, domain):
+        """
+        Fetch the required ReportConfiguration object
+        :param id_: The id of the ReportConfiguration
+        :param domain: The domain of the ReportConfiguration
+        :return: A ReportConfiguration
+        """
+        if report_config_id_is_static(id_):
+            report_config = StaticReportConfiguration.by_id(id_)
+            if report_config.domain != domain:
+                raise NotFound
+        else:
+            try:
+                report_config = get_document_or_not_found(ReportConfiguration, domain, id_)
+            except DocumentNotFound:
+                raise NotFound
+        return report_config
+
+    def base_urls(self):
+        return [url(
+            r"^(?P<resource_name>%s)/(?P<pk>[\w\d_.-]+)/start/(?P<start>[0-9]+)/limit/(?P<limit>[0-9]+)/$" % self._meta.resource_name,
```

Comment:
I was thinking of start and limit as required parameters, and I though that it would make sense to put required parameters in the url, so that it's more explicit that they are always expected. But in retrospect, they don't really need to be required (we can default to 0 for the start, and something else reasonable for the limit). I think I'll move them to the get params.